Brokerages/GDAX/GDAXBrokerage.Messaging.cs

```diff
@@ -226,19 +229,32 @@ private void OrderMatch(string data)
                 direction = message.Side == "sell" ? OrderDirection.Sell : OrderDirection.Buy;
             }
 
+            decimal totalOrderFee;
+            if (!_orderFees.TryGetValue(orderId, out totalOrderFee))
+            {
+                totalOrderFee = GetFee(order);
+                _orderFees[orderId] = totalOrderFee;
+            }
+
+            // apply order fee on a pro rata basis
+            var orderFee = totalOrderFee * Math.Abs(message.Size) / Math.Abs(order.Quantity);
```

Comment:
I think this is great for now, but just considered something -- someone edits the quantity of an order _after_ we've received a partial fill. This would cause the _total_ order feed to change as well as the pro rata calculations.

I'm not considering we attempt to address this now, just thinking out loud and thinking that there are even more complicated cases that we may have not yet anticipated.

Reply:
I don't think you can update orders in GDAX; agree with your point that Caching the order id's in the brokerage is a bad pattern -- unfortunately its a pattern we've replicated on all the brokerages now.